sled-agent/src/bootstrap/maghemite.rs

```diff
@@ -25,32 +25,33 @@ pub enum Error {
 
 pub async fn enable_mg_ddm_service(
     log: Logger,
-    interface: AddrObject,
+    interfaces: Vec<AddrObject>,
 ) -> Result<(), Error> {
     tokio::task::spawn_blocking(|| {
-        enable_mg_ddm_service_blocking(log, interface)
+        enable_mg_ddm_service_blocking(log, interfaces)
     })
     .await?
 }
 
 fn enable_mg_ddm_service_blocking(
     log: Logger,
-    interface: AddrObject,
+    interfaces: Vec<AddrObject>,
 ) -> Result<(), Error> {
     // TODO-correctness Should we try to shut down / remove any existing mg-ddm
     // service first? This appears to work fine as-is on a restart of the
     // sled-agent service.
     info!(log, "Importing mg-ddm service"; "path" => MANIFEST_PATH);
     smf::Config::import().run(MANIFEST_PATH)?;
 
-    // TODO-cleanup mg-ddm supports multiple interfaces, but `smf` currently
-    // doesn't expose an equivalent of `svccfg addpropvalue`. If we need
-    // multiple interfaces we'll need to extend smf.
-    let interface = interface.to_string();
-    info!(log, "Setting mg-ddm interface"; "interface" => interface.as_str());
+    let interface_names: Vec<String> = interfaces
+        .iter()
+        .map(|interface| format!(r#""{}""#, interface))
+        .collect();
```

Comment:
This has to allocate a temporary string for each interface and then a temporary vec to hold them all; unfortunately implementing this without all those temporary allocations is uglier (although I'd love an equivalently-clear way to do this!). An attempt that only allocates one `String` that it continues to append to:

```rust
    let mut interface_names_iter = interfaces.iter();
    let mut property_value = format!(
        r#"("{}""#,
        interface_names_iter
            .next()
            .expect("cannot enable mg-ddm with empty interface list")
    );
    for interface in interface_names_iter {
        write!(property_value, r#" "{}""#, interface).unwrap();
    }
    property_value.push(')');
```

Given how infrequently this runs and that we expect the number of interfaces to be low (I assume), I don't think this is worth it. But I already wrote it down so I'll leave the comment in case either of those assumptions is wrong. :sweat_smile: